static/src/javascripts/__flow__/types/ab-tests.js

```diff
@@ -8,6 +8,7 @@ declare type Variant = {
     success?: ListenerFunction,
     options?: Object,
     engagementBannerParams?: () => Promise<EngagementBannerParams>,
+    deploymentRules?: DeploymentRules,
```

Comment:
I wonder if there's anything to be gained from subtyping `Variant` with `EpicVariant` and `EngagementBannerVariant` and then making these properties mandatory in their respective types

Reply:
If we did that, maybe we could stop using untyped `options` as a repository for anything & everything and create top-level properties for all the epic-specific bits

Reply:
I agree, how about I merge this one and do a refactor in a new PR.
I definitely want to get rid of `options?: Object,`

Reply:
Also, presumably `deploymentRules` would be useful for the banner as well?